`MissingFieldException` when trying to get transaction receipt

When trying to get transaction receipt via GatewayProvider I get the following exception. It seems that `actualFee` should be an optional parameter

Call: `provider.getTransactionReceipt(txHash)`
Response:
```
 <-- 200 https://alpha4-2.starknet.io/feeder_gateway/get_transaction_receipt?transactionHash=0x334da4f63cc6309ba2429a70f103872ab0ae82cf8d9a73b845184a4713cada5
 content-type: application/json; charset=utf-8
 content-length: 150
 date: Mon, 21 Nov 2022 13:18:28 GMT
 server: Python/3.9 aiohttp/3.8.1
 via: 1.1 google
 alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
{"status": "RECEIVED", "transaction_hash": "0x334da4f63cc6309ba2429a70f103872ab0ae82cf8d9a73b845184a4713cada5", "l2_to_l1_messages": [], "events": []}
```
Exception:
`Caused by: kotlinx.serialization.MissingFieldException: Field 'actualFee' is required for type with serial name 'com.swmansion.starknet.data.types.transactions.GatewayTransactionReceipt', but it was missing`

--- a/lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/TransactionReceipt.kt
+++ b/lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/TransactionReceipt.kt
@@ -34,7 +34,7 @@ enum class TransactionStatus {
 @Serializable
 sealed class TransactionReceipt {
     abstract val hash: Felt
-    abstract val actualFee: Felt
+    abstract val actualFee: Felt?
     abstract val isPending: Boolean
     abstract val type: TransactionReceiptType
     abstract val status: TransactionStatus
@@ -75,7 +75,7 @@ data class GatewayTransactionReceipt(
     override val hash: Felt,
 
     @JsonNames("actual_fee")
-    override val actualFee: Felt,
+    override val actualFee: Felt? = null,
 
     @JsonNames("block_hash")
     override val blockHash: Felt? = null,

--- a/lib/src/test/kotlin/starknet/account/StandardAccountTest.kt
+++ b/lib/src/test/kotlin/starknet/account/StandardAccountTest.kt
@@ -202,7 +202,7 @@ class StandardAccountTest {
 
         val receipt = provider.getTransactionReceipt(result.transactionHash).send()
 
-        assertTrue(receipt.actualFee < maxFee)
+        assertTrue(receipt.actualFee!! < maxFee)
     }
 
     @ParameterizedTest

--- a/lib/src/test/kotlin/starknet/provider/ProviderTest.kt
+++ b/lib/src/test/kotlin/starknet/provider/ProviderTest.kt
@@ -746,7 +746,7 @@ class ProviderTest {
 
     @Test
     fun `get sync information node synced`() {
-        val mocked_response = """
+        val mockedResponse = """
         {
             "id": 0,
             "jsonrpc": "2.0",
@@ -761,7 +761,7 @@ class ProviderTest {
         }
         """.trimIndent()
         val httpService = mock<HttpService> {
-            on { send(any()) } doReturn HttpResponse(true, 200, mocked_response)
+            on { send(any()) } doReturn HttpResponse(true, 200, mockedResponse)
         }
         val provider = JsonRpcProvider(devnetClient.rpcUrl, StarknetChainId.TESTNET, httpService)
         val request = provider.getSyncing()
@@ -775,5 +775,36 @@ class ProviderTest {
         assertEquals(1, response.currentBlockNumber)
         assertEquals(Felt.fromHex("0x10"), response.highestBlockHash)
         assertEquals(10, response.highestBlockNumber)
+    }
+
+    @Test
+    fun `received gateway receipt`() {
+        // There is no way for us to recreate this behaviour as devnet processes txs right away
+        val httpService = mock<HttpService> {
+            on { send(any()) } doReturn HttpResponse(
+                true,
+                200,
+                """
+                {
+                    "status": "RECEIVED", 
+                    "transaction_hash": "0x334da4f63cc6309ba2429a70f103872ab0ae82cf8d9a73b845184a4713cada5", 
+                    "l2_to_l1_messages": [], 
+                    "events": []
+                }
+                """.trimIndent(),
+            )
+        }
+        val provider = GatewayProvider.makeTestnetClient(httpService)
+        val receipt = provider.getTransactionReceipt(
+            Felt.fromHex("0x334da4f63cc6309ba2429a70f103872ab0ae82cf8d9a73b845184a4713cada5"),
+        ).send() as GatewayTransactionReceipt
+
+        assertEquals(Felt.fromHex("0x334da4f63cc6309ba2429a70f103872ab0ae82cf8d9a73b845184a4713cada5"), receipt.hash)
+        assertEquals(TransactionStatus.PENDING, receipt.status)
+        assertEquals(emptyList<MessageToL1>(), receipt.messagesToL1)
+        assertEquals(emptyList<Event>(), receipt.events)
+        assertNull(receipt.messageToL2)
+        assertNull(receipt.actualFee)
+        assertNull(receipt.failureReason)
     }
 }

--- a/lib/src/test/kotlin/starknet/provider/ProviderTest.kt
+++ b/lib/src/test/kotlin/starknet/provider/ProviderTest.kt
@@ -779,6 +779,7 @@ class ProviderTest {
 
     @Test
     fun `received gateway receipt`() {
+        val hash = Felt.fromHex("0x334da4f63cc6309ba2429a70f103872ab0ae82cf8d9a73b845184a4713cada5")
         // There is no way for us to recreate this behaviour as devnet processes txs right away
         val httpService = mock<HttpService> {
             on { send(any()) } doReturn HttpResponse(
@@ -795,11 +796,9 @@ class ProviderTest {
             )
         }
         val provider = GatewayProvider.makeTestnetClient(httpService)
-        val receipt = provider.getTransactionReceipt(
-            Felt.fromHex("0x334da4f63cc6309ba2429a70f103872ab0ae82cf8d9a73b845184a4713cada5"),
-        ).send() as GatewayTransactionReceipt
+        val receipt = provider.getTransactionReceipt(hash).send() as GatewayTransactionReceipt
 
-        assertEquals(Felt.fromHex("0x334da4f63cc6309ba2429a70f103872ab0ae82cf8d9a73b845184a4713cada5"), receipt.hash)
+        assertEquals(hash, receipt.hash)
         assertEquals(TransactionStatus.PENDING, receipt.status)
         assertEquals(emptyList<MessageToL1>(), receipt.messagesToL1)
         assertEquals(emptyList<Event>(), receipt.events)

